### PR TITLE
Preserve custom semantic tags when changing logical type

### LINKED ISSUE #1130: Setting new logical type with set_types drops user set semantic tags
Setting a new logical type with `set_types` can cause previously set semantic tags to be lost. This behavior seems a bit counter-intuitive when users do not specify new tags. If users do not specify new values for semantic tags during the call to `set_types` any previously set semantic tags should be retained.

### Current Behavior
Custom tag on `vals` column is lost
```python
import pandas as pd
import woodwork as ww

df = pd.DataFrame({'id': [0, 1, 2], 'vals': [True, False, True]})
df.ww.init(semantic_tags={'vals': 'custom_tag'})

df.ww
```
```
       Physical Type Logical Type Semantic Tag(s)
Column                                           
id             int64      Integer     ['numeric']
vals            bool      Boolean  ['custom_tag']
```
```python
df.ww.set_types({'vals': 'boolean_nullable'})
df.ww
```
```
       Physical Type     Logical Type Semantic Tag(s)
Column                                               
id             int64          Integer     ['numeric']
vals         boolean  BooleanNullable              []
```

### Expected Behavior
Custom tag on `vals` column is retained
```python
import pandas as pd
import woodwork as ww

df = pd.DataFrame({'id': [0, 1, 2], 'vals': [True, False, True]})
df.ww.init(semantic_tags={'vals': 'custom_tag'})

df.ww
```
```
       Physical Type Logical Type Semantic Tag(s)
Column                                           
id             int64      Integer     ['numeric']
vals            bool      Boolean  ['custom_tag']
```
```python
df.ww.set_types({'vals': 'boolean_nullable'})
df.ww
```
```
       Physical Type     Logical Type Semantic Tag(s)
Column                                               
id             int64          Integer     ['numeric']
vals         boolean  BooleanNullable  ['custom_tag']
```

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #1130

### CODE REVIEW

File: woodwork/tests/schema/test_table_schema.py
Comment (thehomebrewnerd) [inline woodwork/tests/schema/test_table_schema.py:578]: Is this the behavior we want? I was thinking that if a user specifies a new set of tags to use any existing tags should get cleared out. The only time we would leave existing tags alone is when `semantic_tags=None` in the call to `set_types`. If users want to add tags to existing tags they would use the `add_semantic_tags` method.

Comment (thehomebrewnerd) [reply woodwork/tests/schema/test_table_schema.py:578]: If you haven't started updating already, it might not hurt to get a second opinion to clarify how we want this to work before you spend time updating things you have already done.

Comment (rwedge) [reply woodwork/tests/schema/test_table_schema.py:578]: +1 for `semantic_tags=set()` meaning remove all tags and `semantic_tags=None` mean preserve custom tags

File: woodwork/column_schema.py
Comment (tamargrey) [inline woodwork/column_schema.py:222]: you can add this to the api reference

File: woodwork/tests/accessor/test_table_accessor.py
Comment (tamargrey) [inline woodwork/tests/accessor/test_table_accessor.py:1966]: This assertion should probably go above line 1958, since we're specifically checking that changing the logical type for a column doesnt remove the custom semantic tag.

File: woodwork/tests/schema/test_table_schema.py
Comment (tamargrey) [inline woodwork/tests/schema/test_table_schema.py:558]: update this comment to make it clear that we dont want the custom tags to be reset

File: woodwork/tests/schema/test_table_schema.py
Comment (jeff-hernandez) [reply woodwork/tests/schema/test_table_schema.py]: updated

**Review by thehomebrewnerd — COMMENTED** [review]
Just a couple really minor things, but otherwise looks good to me.

File: docs/source/release_notes.rst
Comment (jeff-hernandez) [reply docs/source/release_notes.rst]: updated

File: woodwork/table_schema.py
Comment (jeff-hernandez) [reply woodwork/table_schema.py:243]: updated

File: docs/source/release_notes.rst
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:28]: This needs to be in the Future Release section now

**Review by thehomebrewnerd — APPROVED** [review]
LGTM, assuming tests pass!

**Pull request merged**
